### ISSUE TITLE
Check if class uses an aliases only on extends or implements

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#147]
Aliases could help you to workaround a class name conflict when your class have the same name of the super class. But if you rename your class, then this aliases are not more need. For instance:

```php
use MyClass as SuperClass;

class MyClass extends SuperClass {
}
```

But if you rename:

```php
use MyClass as SuperClass;

class NewClass extends SuperClass {
}
```

Is better you remove the alias and use the direct class (autofix is welcome).

```php
class NewClass extends MyClass {
}
```

**Comment by funivan** [comment]
This action can be applied only after user rename. I do not know if it is possible to catch up this action in the phpstorm. Probably this is out of the scope and it will be hard to implement.

**Comment by rentalhost** [comment]
It'll depends if you will rename it by PHPStorm or manually. In all cases, I think that the best option is check normally like others inspections. Not seems to be hard to implements it, basically check if current class is using an aliases, then check if the reason for that is because it have same names.

**Comment by funivan** [comment]
@rentalhost if user rename class manually then inspection should not fire, and if user rename class with PhpStorm then inspection should trigger error? 
What to do if user rename class with phpstorm, and restart it after some minutes?

**Comment by kalessil** [comment]
Reopening

**Comment by kalessil** [comment]
Revisiting, I tend to use the following approach:

```php
namespace NS\Application;
use NS\Domain\Clazz as ClazzContract;

class Clazz implements ClazzContract {
}
```

So wondering if the idea was to check only extension, or implementation constructs as well. @rentalhost : can you clarify this, please?

**Comment by kalessil** [comment]
Re-opening

**Comment by rentalhost** [comment]
@kalessil I have created a new label called "in review" that you could use to avoid stalebot to close automatically. So you can apply to issues that should be always opened until it be implemented.

**Comment by kalessil** [comment]
@rentalhost: oh, thanks. I'll update the stalebot today accordingly.

**Comment by ea-inspections-team** [comment]
Thanks for reporting @rentalhost ! This case is typically a re-modeling/refactoring scenario (90% intentional one) and @kalessil also references to one of the valid cases. Let's skip this one and look for more items in the backlog.
